File: PalindromeChecker/palindrome_checker.py
from collections import deque

#initialize data structures
stack = []
queue = deque([])
stack_out = ""
queue_out = ""

#prompt user for input
to_check = input("Enter item to check: ")

#loop to populate stack and queue
for i in range(len(to_check)):
    stack.append(to_check[i].lower())
    queue.append(to_check[i].lower())

#empty stack
while len(stack) > 0:
    stack_out += stack.pop()

#empty queue
while len(queue) > 0:
    queue_out += queue.popleft()

#check if palindrome
if queue_out == stack_out:
    print(to_check + " is a palindrome!")
else:
    print(to_check + " is not a palindrome.")

File: PalindromeChecker/PalindromeChecker.java
package palindromechecker;

import java.util.ArrayDeque;
import java.util.Scanner;
import java.util.Stack;

public class PalindromeChecker {
  
    public static void main(String[] args) {
        // initialize data structures
        Scanner scan = new Scanner(System.in);
        String toCheck = new String();        
        Stack stack = new Stack();
        ArrayDeque queue = new ArrayDeque();
        String stackOut, queueOut = new String();
        StringBuilder temp1 = new StringBuilder();
        StringBuilder temp2 = new StringBuilder();
        
        //prompt for user input 
        System.out.println("Enter item to check: ");
        toCheck = scan.nextLine();
        
        //loop to put each character into stack and queue
        for (int i=0; i < toCheck.length(); i++) {
            stack.push(Character.toLowerCase(toCheck.charAt(i)));
            queue.add(Character.toLowerCase(toCheck.charAt(i)));
        }
        
        //empty stack        
        while (!stack.empty()) {
            temp1.append(stack.pop());
        }
        stackOut = temp1.toString();
        
        //empty the queue
        while (!queue.isEmpty()) {
            temp2.append(queue.removeFirst());
        }
        queueOut = temp2.toString();
        
        //check if palindrome
        if(queueOut.equals(stackOut)) {
            System.out.println(toCheck + " is a palindrome!");
        }
        else {
            System.out.println(toCheck + " is not a palindrome.");
        }
    }    
}
